src/payment/strategies/square/square-form.ts

```diff
@@ -19,6 +19,60 @@ export interface SquareFormOptions {
     cvv: SquareFormElement;
     expirationDate: SquareFormElement;
     postalCode: SquareFormElement;
+    masterpass: SquareFormElement;
+}
+
+export interface NonceGenerationError {
+    type: string;
+    message: string;
+    field: string;
+}
+
+export interface SquareValidationErrors {
+    country: string[];
+    region: string[];
+    city: string[];
+    addressLines: string[];
+    postalCode: string[];
+    [key: string]: string[];
+}
+
+export interface CardData {
+    card_brand: CardBrand;
+    last_4: number;
+    exp_month: number;
+    exp_year: number;
+    billing_postal_code: string;
+    digital_wallet_type: DigitalWalletType;
+}
+
+export interface Contact {
+    familyName: string;
+    givenName: string;
+    email: string;
+    country: string;
+    countryName: string;
+    region: string;
+    city: string;
+    addressLines: string[];
+    postalCode: string;
+}
+
+export enum CardBrand {
```

Comment:
What values do you expect to get when you send these enums to the API?

Enums in TypeScript, unless specified, will be mapped to integers from `0 .. length`. So in this case, using `CardBrand.americanExpress` will return `0`, `CardBrands.discover` will be 1, and so on and so forth.

If that's the expected behaviour, you don't have to change anything. If you expected strings, you can just add them in the enum definition:

```js
enum CardType {
    americanExpress = 'AMERICAN_EXPRESS',
    discover = 'DISCOVER',
    discoverDiners,
    JCB,
    masterCard,
    unionPay,
    unknown,
    visa,
}
```

Reply:
👍

Reply:
What values do you expect to get when you send these enums to the API?

Reply:
actually is the other way around [these values are being returned from the API](https://docs.connect.squareup.com/api/connect/v2#type-cardbrand)

Reply:
Oh, cool. Better to use the correct values then.

Reply:
@icatalina 90.57 % of Coverage